app/controllers/ConfirmationController.scala

```diff
@@ -37,13 +38,20 @@ class ConfirmationController @Inject()(
                                         override val requireData: DataRequiredAction,
                                         override val userAllowList: UserAllowListAction,
                                         val controllerComponents: MessagesControllerComponents,
-                                        view: ConfirmationView
+                                        view: ConfirmationView,
+                                        errorHandler: ErrorHandler
                                       ) extends BaseNavigationController with AuthActionHelper {
 
   def onPageLoad(ern: String, arc: String): Action[AnyContent] =
-    authorisedDataRequestWithCachedMovementAsync(ern, arc) { implicit request =>
-      withAnswer(CheckYourAnswersPage) { confirmationDetails =>
-        Future.successful(Ok(view(confirmationDetails)))
+    authorisedDataRequestWithCachedMovement(ern, arc) { implicit request =>
+      request.userAnswers.get(ConfirmationPage) match {
+        case Some(confirmationDetails: ConfirmationDetails) =>
+          logger.info("[onPageLoad] Successful Explain Delay confirmation page displayed")
+          Ok(view(confirmationDetails))
+        case None =>
+          logger.warn("[onPageLoad] Could not retrieve submission confirmation details from User session")
+          Redirect(routes.JourneyRecoveryController.onPageLoad(ern, arc))
```

Comment:
Does `withAnswer` not achieve the same thing here and keep it cleaner?

Reply:
Pushed a commit to apply that change https://github.com/hmrc/emcs-tfe-explain-delay-frontend/pull/11/commits/10074fe45d2fb5be34ac31ba1013a42ad5edd443